### PR TITLE
ElementType.TYPE added to the ApiOperation, Test Added

### PR DESCRIPTION
**Pull request by ralphdoe**
Solution for the issue: 
https://github.com/swagger-api/swagger-core/issues/2652

### CODE REVIEW

File: modules/swagger-servlet/src/main/java/io/swagger/servlet/extensions/ServletReaderExtension.java [hunk 1]
```diff
@@ -405,8 +404,8 @@ public void applyImplicitParameters(ReaderContext context, Operation operation,
 
     @Override
     public void applyExtensions(ReaderContext context, Operation operation, Method method) {
-        final ApiOperation apiOperation = method.getAnnotation( ApiOperation.class );
-        if( apiOperation != null ) {
+        final ApiOperation apiOperation = method.getAnnotation(ApiOperation.class);
+        if (apiOperation != null) {
             operation.getVendorExtensions().putAll(BaseReaderUtils.parseExtensions(apiOperation.extensions()));
         }
     }
```
Comment (frantuma) [inline modules/swagger-servlet/src/main/java/io/swagger/servlet/extensions/ServletReaderExtension.java]: All these cleanup should be part of separate PR

File: modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+public class ApiOperationTypeTest {
+
+    @Test
+    public void testApiOperationInType() {
+
+        final Swagger swagger = new Swagger();
+        Reader.read(swagger, Collections.<Class<?>>singleton(ApiOperationResource.class));
+        Assert.assertNotNull(swagger.getPaths().get("/users"));
+    }
+}
```
Comment (frantuma) [inline modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java]: test should be part of swagger-jaxrs, not swagger-servlet (or even better have them both)

also where does the path "users" come from?

File: modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/ApiOperationResource.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package io.swagger.servlet.resources;
+
+public class ApiOperationResource {
+
+    @Uber(description = "Description")
+    public String getResponse() {
+        return "Response";
+    }
+}
```
Comment (frantuma) [inline modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/ApiOperationResource.java]: test should be part of swagger-jaxrs, not swagger-servlet (or even better have them both)

File: modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/Uber.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        extensions = {
+                @Extension(properties = {
+                        @ExtensionProperty(name = "name", value = "value")
+                })}
+)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Uber {
+    String description() default "";
+
+}
```
Comment (frantuma) [inline modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/Uber.java]: test should be part of swagger-jaxrs, not swagger-servlet (or even better have them both)

File: modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java (hunk 2 above)
Comment (ralphdoe) [reply modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java]: It has been extracted from the:

```
@ApiOperation(
        value = "Meta Uber Annotation",
        notes = "Test description",
        consumes = "multipart/form-data",
        produces = "multipart/form-data",
        httpMethod = "GET",
        *nickname = "users",*
```

Comment (ralphdoe) [reply modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java]: Test added to JaxRS module

File: modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/ApiOperationResource.java (hunk 3 above)
Comment (ralphdoe) [reply modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/ApiOperationResource.java]: Test added to JaxRS module

File: modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/Uber.java (hunk 4 above)
Comment (ralphdoe) [reply modules/swagger-servlet/src/test/java/io/swagger/servlet/resources/Uber.java]: Test added to JaxRS module

File: modules/swagger-servlet/src/main/java/io/swagger/servlet/extensions/ServletReaderExtension.java (hunk 1 above)
Comment (ralphdoe) [reply modules/swagger-servlet/src/main/java/io/swagger/servlet/extensions/ServletReaderExtension.java]: I'm going to push this in another PR.

File: modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+public class ApiOperationTypeTest {
+
+    private static final String USERS_PATH = "/users";
+
+    @Test
+    public void testApiOperationInType() {
+        final Swagger swagger = new Swagger();
+        Reader.read(swagger, Collections.<Class<?>>singleton(ApiOperationResource.class));
+        Assert.assertNotNull(swagger.getPaths().get(USERS_PATH));
```
Comment (frantuma) [inline modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java]: This doesn't really test the `ApiOperation` annotation..

Comment (ralphdoe) [reply modules/swagger-servlet/src/test/java/io/swagger/servlet/extensions/ApiOperationTypeTest.java]: Renamed to testThatApiOperationMightBeInType.
